Classes/Event/EventBus.php

```diff
@@ -43,18 +44,13 @@ class EventBus
     public function handle(EventTransport $transport)
     {
         $listeners = $this->locator->getListeners($transport->getEvent());
-
-        if ($listeners === null) {
-            return;
-        }
-
-        /** @var EventListenerContainer $listener */
+        /** @var \callable $listener */
         foreach ($listeners as $listener) {
             try {
-                $listener->when($transport);
+                call_user_func($listener, $transport->getEvent(), $transport->getMetadata());
```

Comment:
`call_user_func` seems odd, but it will work no matter if the listener is a `\Closure` some array in the form `[$object, 'methodName']`, a string `ClassName::methodName` or any other [callable](http://php.net/manual/en/language.types.callable.php)